apps/api/src/app/events/usecases/trigger-event/message-matcher.service.ts

```diff
@@ -1,29 +1,26 @@
-import {
-  NotificationStepEntity,
-  SubscriberEntity,
-  EnvironmentRepository,
-  SubscriberRepository,
-  JobEntity,
-} from '@novu/dal';
-import { ITriggerPayload } from '@novu/node';
 import * as _ from 'lodash';
 import { createHmac } from 'crypto';
+import axios from 'axios';
+import { Injectable } from '@nestjs/common';
+import type { ITriggerPayload } from '@novu/node';
+import type { IBaseFieldFilterPart, FilterParts, IWebhookFilterPart, IFieldFilterPart } from '@novu/shared';
+import { SubscriberEntity, EnvironmentRepository, SubscriberRepository, StepFilter } from '@novu/dal';
+
 import { CreateExecutionDetails } from '../../../execution-details/usecases/create-execution-details/create-execution-details.usecase';
 import { SendMessageCommand } from '../send-message/send-message.command';
 import { EXCEPTION_MESSAGE_ON_WEBHOOK_FILTER } from '../../../shared/constants';
-import axios from 'axios';
-import { Injectable } from '@nestjs/common';
 
 export interface IFilterVariables {
-  payload: ITriggerPayload;
+  payload?: ITriggerPayload;
   subscriber?: SubscriberEntity;
   webhook?: Record<string, unknown>;
 }
 
-export interface IMessageFilterConfiguration {
-  job: JobEntity;
-  command: SendMessageCommand;
-}
+const findFilterPartByType = (filterParts: FilterParts[], type: FilterParts['on'] | FilterParts['on'][]) => {
+  const types = Array.isArray(type) ? type : [type];
+
+  return filterParts.filter((filterPart) => types.includes(filterPart.on));
+};
```

Comment:
Added this helper which allows finding all the filters by type. It will be useful for the next PR when we will introduce the online filters implementation